File: completed/2461.java
import java.util.*;

class Solution {
    public long maximumSubarraySum(int[] nums, int k) {
        long maximum = 0;
        long currentSum = 0;
        Map<Integer, Integer> currentNums = new HashMap<>();
        for (int i = 0; i < k; i++) {
            currentSum += nums[i];
            currentNums.put(nums[i], currentNums.getOrDefault(nums[i], 0) + 1);
        }
        if (currentNums.keySet().size() == k) {
            maximum = currentSum;
        }

        for (int i = k; i < nums.length; i++) {
            currentSum -= nums[i - k];
            int count = currentNums.get(nums[i - k]);
            if (count == 1) {
                currentNums.remove(nums[i - k]);
            } else {
                currentNums.put(nums[i - k], count - 1);
            }

            currentSum += nums[i];
            currentNums.put(nums[i], currentNums.getOrDefault(nums[i], 0) + 1);
            if (currentNums.keySet().size() == k) {
                maximum = Math.max(currentSum, maximum);
            }
        }
        return maximum;
    }
}

File: completed/2461.swift
import Collections

class Solution {
    func maximumSubarraySum(_ nums: [Int], _ k: Int) -> Int {
        var seen = Set<Int>()
        var left = 0
        var right = 0
        var maximum = 0
        var current = 0
        while right < nums.count {
            while right - left >= k || seen.contains(nums[right]) {
                seen.remove(nums[left])
                current -= nums[left]
                left += 1
            }

            seen.insert(nums[right])
            current += nums[right]
            if seen.count == k {
                maximum = max(current, maximum)
            }
            right += 1
        }
        return maximum
    }
}
